sbi/utils/sbiutils.py

```diff
@@ -444,29 +446,35 @@ def logit(theta: Tensor, lower_bound: Tensor, upper_bound: Tensor) -> Tensor:
     return torch.log(theta_01 / (1 - theta_01))
 
 
-def check_if_boxuniform(dist) -> Tuple[bool, Optional[BoxUniform]]:
-    """Returns whether the `dist` is `BoxUniform` as well as the `BoxUniform` itself.
+def check_if_uniform_or_gaussian(dist) -> Tuple[bool, Optional[Distribution]]:
+    """Returns whether the `dist` is `Uniform`.
 
-    When the user called `prepare_for_sbi`, the distribution will in fact be a
+    The dist can be hidden in an Independent distribution, a Boxuniform or in a wrapper.
+    E.g., when the user called `prepare_for_sbi`, the distribution will in fact be a
     `PytorchReturnTypeWrapper`. Thus, we need additional checks.
 
     Args:
         dist: Distribution to be checked.
 
     Returns:
-        Whether the `dist` is `BoxUniform` and the `BoxUniform` itself.
+        Whether the `dist` is `Uniform` and the `Uniform` itself.
     """
-    if isinstance(dist, BoxUniform):
-        is_boxuniform = True
-        box_dist = dist
-    elif hasattr(dist, "prior") and isinstance(dist.prior, BoxUniform):
-        is_boxuniform = True
-        box_dist = dist.prior
```

Comment:
Did you make sure that these checks are really not needed if you run `prepare_for_sbi` on the prior? It might get wrapped in `PytorchDistributionWrapper` and thus you need the `elif` case.

Reply:
Yes, I checked it. And it makes sense because we implicitly check for `BoxUniform(Independent)`